tempodb/encoding/vparquet/schema.go

```diff
@@ -563,3 +560,13 @@ func parquetTraceToTempopbTrace(parquetTrace *Trace) *tempopb.Trace {
 
 	return protoTrace
 }
+
+func extendReuseSlice[T any](sz int, in []T) []T {
+	// jpe nil out slice to avoid leaking memory
+	if cap(in) >= sz {
+		// slice is large enough
+		return in[:sz]
```

Comment:
Does `len(in)` end up being equal to `len(sz)`?  If so, doesn't that leave the original data at the positions?

Reply:
yup, good thought. the assumption is that the calling code will overwrite every single field on the structs in the slice. if every field is not overwritten then, yes, data from a different trace could "leak" through.

Reply:
Good call. There were definitely some fields that were not zero'ed out. Added.